src/main/java/core/basesyntax/supplier/ShapeSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.shape.Circle;
+import core.basesyntax.shape.IsoscelesTrapezoid;
+import core.basesyntax.shape.Rectangle;
+import core.basesyntax.shape.RightTriangle;
+import core.basesyntax.shape.Shape;
+import core.basesyntax.shape.Square;
+import core.basesyntax.type.Color;
+
+public class ShapeSupplier extends BasicRandomValueSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    @Override
+    public Shape getRandom() {
+        Color color = colorSupplier.getRandom();
+        double randomDouble1 = randomGenerator.nextDouble();
+        double randomDouble2 = randomGenerator.nextDouble();
+        double randomDouble3 = randomGenerator.nextDouble();
+
+        int randomShapePick = randomGenerator.nextInt(Shape.SHAPES_COUNT);
```

Comment:
With such approach you create new Random object each time you want to get a number.
This is not good, please make Random class field and use in same manner as ColorSupplier

private final Random random = new Random();

Same should be done for ColorSupplier class. After that you may remove your BasicRandomValueSupplier class

Reply:
Sorry, i'm not sure if i understand this comment right, i checked the hash code for `randomGenerator` for both suppliers and it stays the same within the method calls. Could you please explain me more?

```
class ColorSupplier {
    getRandom() {
        System.out.println(randomGenerator.hashCode());
        ...
        return ...
    }
}

class ShapeSupplier {
    ColorSupplier colorSupplier = new ColorSupplier();
    
    getRandom() {
        System.out.println(randomGenerator.hashCode());
        ...
        return ...
    }
}

class Main {
    main() {
        ShapeSupplier shapeSupplier = new ShapeSupplier();
        
        shapeSupplier.getRandom();
        // 1804094807  <- ColorSupplier's randomGenerator hash code
        // 2001049719  <- ShapeSupplier's randomGenerator hash code
        
        shapeSupplier.getRandom();
        // 1804094807  <- ColorSupplier's randomGenerator hash code
        // 2001049719  <- ShapeSupplier's randomGenerator hash code
        
        shapeSupplier.getRandom();
        // 1804094807  <- ColorSupplier's randomGenerator hash code
        // 2001049719  <- ShapeSupplier's randomGenerator hash code
    }
}
```

Reply:
Sorry, my mistake, on first check it seemed for me that BasicRandomValueSupplier contained method that provides new Random() for child classes.
But it's actually just parent class field and sure it's fine and will provide same object for all method calls in child classes